airbyte-integrations/connectors/source-facebook-marketing/source_facebook_marketing/client/api.py

```diff
@@ -156,8 +156,10 @@ def read(self, getter: Callable, params: Mapping[str, Any] = None) -> Iterator:
         latest_cursor = None
         for record in super().read(getter, params):
             cursor = pendulum.parse(record[self.state_pk])
-            if self._state and self._state >= cursor:
-                continue
+            if self._state:
+                buffer_days = getattr(self, 'buffer_days') + 1 if hasattr(self, 'buffer_days') else 0
```

Comment:
(nitpick) We can declare buffer_days before the loop, and simply replace `if self._state and self._state >= cursor:` with `if self._state and self._state.subtract(days=buffer_days) >= cursor:`. It will look prettier, WDYT?

Reply:
Yes, we can move declaring `buffer_days` from the loop. But we should consider that sometimes self._state is None.